app/src/main/java/tech/ula/ui/SessionListFragment.kt

```diff
@@ -135,173 +156,107 @@ class SessionListActivity : AppCompatActivity() {
-                    ActivityCompat.requestPermissions(this, arrayOf(
+                    ActivityCompat.requestPermissions(activityContext, arrayOf(
                             Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE),
                             permissionRequestCode)
                     dialog.dismiss()
-                })
-                .setNegativeButton(R.string.alert_permissions_necessary_settings_button, {
+                }
+                .setNegativeButton(R.string.alert_permissions_necessary_settings_button) {
                     dialog, _ ->
                     val settingsIntent = Intent(
                             android.provider.Settings.ACTION_APPLICATION_DETAILS_SETTINGS,
                             Uri.parse("package:${BuildConfig.APPLICATION_ID}"))
                     startActivity(settingsIntent)
                     dialog.dismiss()
-                })
-                .setNeutralButton(R.string.alert_permissions_necessary_cancel_button, {
+                }
+                .setNeutralButton(R.string.alert_permissions_necessary_cancel_button) {
                     dialog, _ ->
                     dialog.dismiss()
-                })
+                }
         builder.create().show()
     }
 
-
-    private fun deleteSession(session: Session): Boolean {
-        killSession(session)
-        sessionListViewModel.deleteSessionById(session.id)
-        return true
+    override fun onCreateContextMenu(menu: ContextMenu?, v: View?, menuInfo: ContextMenu.ContextMenuInfo?) {
+        super.onCreateContextMenu(menu, v, menuInfo)
+        activityContext.menuInflater.inflate(R.menu.context_menu_sessions, menu)
     }
 
-    private fun navigateToFilesystemManagement(): Boolean {
-        val intent = Intent(this, FilesystemListActivity::class.java)
-        startActivity(intent)
-        return true
+    override fun onContextItemSelected(item: MenuItem): Boolean {
+        val menuInfo = item.menuInfo as AdapterView.AdapterContextMenuInfo
+        val position = menuInfo.position
+        val session = sessionList[position]
+        return when(item.itemId) {
+            R.id.menu_item_session_kill_service -> stopService(session)
+            R.id.menu_item_session_edit -> editSession(session)
+            R.id.menu_item_session_delete -> deleteSession(session)
+            else -> super.onContextItemSelected(item)
+        }
     }
 
-    private fun navigateToSettings(): Boolean {
-        val intent = Intent(this, SettingsActivity::class.java)
-        startActivity(intent)
+    private fun stopService(session: Session): Boolean {
+        if(session.active) {
+            session.active = false
+            sessionListViewModel.updateSession(session)
+            val view = list_sessions.getChildAt(sessionList.indexOf(session))
+            view.image_list_item_active.setImageResource(R.drawable.ic_block_red_24dp)
+
+            serverUtility.stopService(session)
```

Comment:
I believe this call of stopService is redundant with what is done inside ServerService when the type is kill